src/services/classifier.ts

```diff
@@ -0,0 +1,192 @@
+  * @param {number} [startTime] The start time in seconds of our clip.mp4
+  * @param {number} [duration] The duration in seconds of our clip
+  *
+  * @return {Promise<FootageBoundingBoxType[]>} Returns the array of FootageBoundingBoxType which we can then just send to Rhombus to create the bounding boxes on the console
+  * */
+export const ClassifyDirectory = async (model: ObjectDetection, directory: string, startTime: number, duration: number): Promise<FootageBoundingBoxType[]> => {
+	// Get all of the files in our resource directory
+	let files = fs.readdirSync(directory, { withFileTypes: true });
+
+	// If the file is not a JPEG, then we don't want to include it in later processing
+	files = files.filter(file => file.name.search("jpg") != -1);
+
+	// We need to sort our JPEGs alphabetically, so that we make sure they are in order
+	files.sort((a, b) => {
+		if (a.name < b.name) return -1;
+		if (a.name > b.name) return 1;
+		return 0;
+	});
+
+	// Create our array of bounding boxes which will then later be processed
+	let boxes: BoundingBox[] = [];
+
+	// The dimensions of our JPEG. This should theoretically remain constant between every single frame, but we will set it for every frame just in case
+	let dimensions: Vec2;
+
+	// Loop through all of our files
+	for (let i = 0; i < files.length; i++) {
+		// Get the timestamp of our frame, which will be the startTime in seconds + some offset
+		// The offset will just be the index / the number of files * our duration, that way we can see what fraction of the way we are through our clip just by looking at the index
+		// We also will multiply everything by 1000 since the timestamp has to be in ms, and right now we are in seconds
+		const timestamp = (startTime + (i / files.length) * duration) * 1000;
+
+		// Classify our image
+		const [res, dim] = await ClassifyImage(model, directory + files[i].name, timestamp)
+
+		// Add those new results to our boxes
+		boxes = boxes.concat(res);
+
+		// Update the dimensions of our JPEG just in case
+		dimensions = dim;
+	}
+
+	// Create our final array of boxes
+	let boundingBoxes: FootageBoundingBoxType[] = [];
+
+	// Loop through all of our own boxes and convert them to FootageBoundingBoxTypes
+	for (let i = 0; i < boxes.length; i++) {
+		const box = boxes[i];
+
+		// We are using the top left as the bounding box position, so top left is (box.x, box.y), bottom right is (box.x + box.width, box.y + box.height)
+		boundingBoxes.push({
+			a: ActivityEnum.CUSTOM,
+			// These values are permyriads, so we need to convert our bounding boxes appropriately
+			b: (box.position.y + box.dimensions.y) / dimensions.y * 10000, // Bottom
+			l: (box.position.x / dimensions.x) * 10000, // Left
+			r: ((box.position.x + box.dimensions.x) / dimensions.x) * 10000, // Right
+			t: (box.position.y / dimensions.y) * 10000, // Top
+			ts: box.timestamp,
+			cdn: box.label,
+			ro: [box.label],
```

Comment:
this field `ro` is technically optional and is not used in the Rhombus console at this moment.